pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -97,11 +97,11 @@ def get_reduced_metrics(self):
         return self._internals_reduced
 
     def add_dataloader_idx(self):
-        return len(self._internals) > 1
+        return len(self._internals_reduced) > 1 if self.has_reduced else len(self._internals) > 1
 
     @property
     def num_dataloaders(self):
-        return len(self._internals)
+        return len(self._internals_reduced) if self.has_reduced else len(self._internals)
```

Comment:
easier to read
```suggestion
        _inter = self._internals_reduced if self.has_reduced else self._internals
        return len(_inter)
```
pls add typing :]